text/0000-scoped-slots-light-dom.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Scoped Slots in Light DOM
+status: DRAFTED
+created_at: 2022-06-10
+updated_at: 2022-06-22
+pr: 
+---
+
+# Scoped Slots in Light DOM
+
+## Summary
+
+[Scoped slots](https://v3.vuejs.org/guide/component-slots.html#scoped-slots) is a feature from the Vue
+framework, that allows passing data from a child component to the content slotted from a parent component
+in a declarative fashion.
+
+One of the most common use-cases for this is to use a generic component to render an array of items and
+let the parent component customize how each item is rendered. For example a generic infinite list
+component, a virtualized list component or, a data table.
+
+*This proposal is for adding Scoped Slots only to Light DOM.*
+
+## Basic example
+
+```html
+// x/parent.html
+<template> <!-- parent need not be light DOM -->
+    <x-list items={items} lwc:slot-data="item">
+        <span>{item.name}</span>
+    </x-list>
+</template>
+
+// x/list.html
+<template lwc:render-mode="light">
+    <ul>
+        <template for:each={items} for:item="item">
+            <li key={item.id}>
+                <slot lwc:slot-bind={item}></slot> <!-- delegating rendering of children to the parent -->
+            </li>
+        </template>
+    </ul>
+</template>
+```
+The `<x-parent>` component is slotting `<span>{item.name}</span>` into `<x-list>` but the `item` is supplied
+by `<x-list>`, the child component. It's essentially like `<x-parent>` is slotting a function that actually
+returns the slotted content.
+
+This allows `<x-parent>` to control how each item renders while leaving the looping/repeating logic to `<x-list>`.
+`<x-list>` is now free to create as many slots as it wants while using the same "slotting function" passed from the parent.
+This allows `<x-list>` to be a virtual list or an infinite list.
+
+It's important to note that for this to happen, slots need to be lazily rendered.
+
+* Native shadow DOM slots: the slotted content is eagerly rendered regardless of there being a slot to “accept” the content. The component slotting the content is in charge of appending and removing the content from the DOM.
+* Scoped slots: the slotted content is lazily rendered; the slotted content will only be rendered if there is a slot to “accept” the content. The component with the <slot> elements is the one in charge of rendering the slotted content passed by the parent component.

```

Comment:
Something super important to call out here is that for scoped slot the fragment is still "owned" by the parent component. 

This had direct implications in terms of styling. If the parent component has a scoped stylesheet associated with it, the styles also apply to the scoped slot content (like standard slotted content).